### PR TITLE
UI/component refactor part 2

### PR DESCRIPTION
**Pull request by TheoTheDev**
Components refactored and cleaned up:
- DragAndDrop
- LoadingBar
- ObjectPreview
- Stats
- Header.jsx

### CODE REVIEW

**Comment by alisaad673** [comment]
I have tested it, no new bugs there.
Though these bugs are there in master. Can you have a look at these and see if they are related and can be fixed here?
https://github.com/webaverse/app/issues/2680
https://github.com/webaverse/app/issues/2690

**Comment by ohmed** [comment]
@alisaad673 I think it's better to solve this issues in separate PRs and not combine with this one.

**Comment by ohmed** [comment]
Resolved merge issues.

File: src/components/general/drag-and-drop/DragAndDrop.jsx [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+            const app = currentApp;
+            _importApp( app );
+            app.activate();
+            setCurrentApp( null );
+
+        }
+
+    };
+
+    const _mint = ( event ) => {
```
Comment (0reo) [inline src/components/general/drag-and-drop/DragAndDrop.jsx:332]: this currently does nothing?  if it's going to be updated later, can we add a comment here stating that?

**Review by 0reo — COMMENTED** [review]
generally looks good.  1 question about it, and could use an eslinting

File: src/components/general/drag-and-drop/DragAndDrop.jsx (hunk 1 above)
Comment (ohmed) [reply src/components/general/drag-and-drop/DragAndDrop.jsx:332]: Not sure about part of dead code,
I also saw some parts, but don't have enough info about this and only the 'author' of the code knows if the code is not important anymore.

**Comment by ohmed** [comment]
> generally looks good. 1 question about it, and could use an eslinting
So the code style rules inside of UI src folder are different from the general projects.
I didn't add separate eslint not to make and confusions.
But if you ask to use same as in the rest of the codebase I can argue cause general code style is quite unreadable comparing to this one.
If it was my decision I would this UI code style entirely for the whole project. I already have eslint config files [hmm, maybe need edits, cause I remembered I have it for TypeScript], but this needs to be general conversation and probably not just here in UI refactor part 2 PR.

File: src/components/general/drag-and-drop/DragAndDrop.jsx (hunk 1 above)
Comment (0reo) [reply src/components/general/drag-and-drop/DragAndDrop.jsx:332]: ok, we'll leave it for now and find out from the author what the situation is

**Comment by avaer** [comment]
The code improvements are quite nice. However, so far every UI component refactor has caused bugs that slipped through testing, and most refactoring PRs had to be reverted as a result.

Therefore I propose we stop this refactoring work and close these PRs as it's a net loss + cost rather than a benefit.

There are techniques which can be used to refactor without breakage, but historically speaking they have not been used, and every UI refactor PR has been a regression in practice. I think it's reasonable to assume any refactor following in the same vein will have the same problems as the previous PRs, so maybe it's time to develop a rule that we simply are not doing this anymore.

**Comment by avaer** [comment]
Thoughts on the above?

**Comment by avaer** [comment]
No comments, looks like this PR is abandoned for now.

**Pull request closed without merging**